### PR TITLE
Introduced .editorconfig .npmrc and formatted code and readme file.

### PR DESCRIPTION
**Pull request by anusaini**
Introduced .editorconfig
Introduced .npmrc
Formatted code accordingly.
Formatted README file.

### CODE REVIEW

**Comment by anusaini** [comment]
@gabrielcsapo let me know if you have different style preferences.

**Comment by gabrielcsapo** [comment]
@anusaini while I like editor config, I usually just stick with eslint and let my editor pull rules from that. Would you be able to migrate these rules to eslint?

**Comment by anusaini** [comment]
Let me see what I can do.

**Comment by anusaini** [comment]
@gabrielcsapo middle ground? 

Enforcing indentation and whitespace related rule via eslintrc too.

`.editorconfig` and `.eslintrc` have different use case. `lint` in build process; editorconfig is dev-help tool.

File: .npmrc [hunk 1]
```diff
@@ -0,0 +1 @@
+registry=https://registry.npmjs.com/
```
Comment (gabrielcsapo) [inline .npmrc:1]: I think this is redundant as we have the registry information in the package.json

File: .eslintrc [hunk 2]
```diff
@@ -11,6 +11,11 @@
       "error",
       "always"
     ],
-    "no-octal": 0
+    "no-octal": 0,
+    "indent": [
```
Comment (gabrielcsapo) [inline .eslintrc:15]: 👍

File: .editorconfig [hunk 3]
```diff
@@ -0,0 +1,8 @@
+[*]
```
Comment (gabrielcsapo) [inline .editorconfig:1]: I am thinking adding this will also be another file to upkeep. Eslint should be good enough.

Comment (anusaini) [reply .editorconfig:1]: Trust me when I say this: we won't be touching this file ever. It's just to give VS Code something to do than manually doing it.

**Pull request closed without merging**